documentation/customisation.md

```diff
@@ -0,0 +1,139 @@
+
+To start off with, create a `Dockerfile` that resembles the following. 
+
+Please note that certain JARS will go into certain paths. You shouldn't add a Oracle JAR to the `ext/` folder, for example. If you are unsure where a JAR should be added, please get in touch with our support team. 
+
+NOTE: `${APP_HOME}` points to the path `/opt/xebialabs/xl-deploy-server` by default
+
+```docker
+FROM xebialabs/xl-release:9.5.0
+
+###################################################################################
+# PLUGINS                                                                         #
+# Plugins should be placed under ${APP_HOME}/default-plugins/ #
+###################################################################################
+
+COPY --chown=10001:0 files/xld-liquibase-plugin-5.0.1.xldp /opt/xebialabs/xl-release-server/default-plugins/xlr-official/
+
+# Add plugin from url
+ADD --chown=10001:0 https://dist.xebialabs.com/public/community/xl-deploy/command2-plugin/3.9.1-1/command2-plugin-3.9.1-1.jar /opt/xebialabs/xl-release-server/default-plugins/__local__/
+
+##########################################################################
+# EXTENSIONS                                                             #
+# Extensions should be placed under ${APP_HOME}/ext
+##########################################################################
+ADD --chown=10001:0 files/ext /opt/xebialabs/xl-release-server/ext/
+
+##########################################################################
+# HOTFIXES                                                               #
+##########################################################################
+ADD --chown=10001:0 files/hotfix.jar /opt/xebialabs/xl-release-server/hotfix/
+
+##########################################################################
+# LIBRARIES                                                              #
+##########################################################################
+ADD --chown=10001:0 files/ojdbc6.jar /opt/xebialabs/xl-release-server/lib/
+```
+
+NOTE: For XL Release plugins, official plugins should go to the `default-plugins/xlr-official/` folder, while custom or community plugins should go under `default-plugins/__local__/`
+
+For an overview of how `ADD` and `COPY` works, please refer to [the documentation](https://docs.docker.com/engine/reference/builder/#add)
+
+Once you are satisfied with your Dockerfile, run the following command in the same directory
+
+`docker build -t xl-release-custom:9.5.0 .`
+
+Always use [semver](https://semver.org/) to version your docker images. This is to ensure future compatability with one of our other tools, `xl up`.
+
+This command will build and tag a docker image for you. If you would like to run this image locally, you can do this now by running
+
+`docker run -it --rm -p 5601:5601 -e ACCEPT_EULA=Y xl-release-custom:9.5.0`
```

Comment:
- This will not work as XLRelease uses port 5516 not 5601
- Also based on above comment custom image tag should be 9.0
- I think it is better to add Admin password here using -e "ADMIN_PASSWORD=desired_admin_password"

Reply:
indeed:

* This will not work as XLRelease uses port 5516 not 5601
* I think it is better to add Admin password here using -e "ADMIN_PASSWORD=desired_admin_password"
should be fixed